### DISCUSSION TITLE
OnBeforeResourceLoad

### DISCUSSION THREAD

**Discussion by derbuchmann** [cefsharp/CefSharp#3358] [Q&A]
Hi, 
i am using cefsharp 86.0.0 and it is really great. Thank You so much.
But I have a question?
I want to add a request header ( a identifier). I do it on the ResourceRequestHandler.OnBeforeResourceLoad an it works!
But not with YouTube!!!
When I add a header, no video plays anymore!
I also put the headername to the "Access-Control-Request-Headers", but nothing helps.

Can someone maybe help?
Thank you

Jo A. Kim

**Comment by amaitland** [comment]
The short version is only add headers to sites that are within your control or that you've tested and have confirmed support handling pre flight requests. 

See https://github.com/cefsharp/CefSharp/issues/3277 for further details. I'd also suggest reading https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests
